salt/defaults/__init__.py

```diff
@@ -5,5 +5,58 @@
 as this may result in circular imports.
 """
 
+
+class _Constant:
+    """
+    This class implements a way to create constants in python.
+
+    NOTE:
+
+      - This is not really a constant, ie, the `is` check will not work, you'll
+        have to use `==`.
+      - This class SHALL NOT be considered public API and might change or even
+        go away at any given time.
+    """
+
+    __slots__ = ("name", "value")
+
+    def __init__(self, name, value=None):
+        self.name = name
+        self.value = value
+
+    def __hash__(self):
+        return hash((self.name, self.value))
+
+    def __eq__(self, other):
+        if not isinstance(other, self.__class__):
+            return False
+        if self.name != other.name:
+            return False
+        return self.value == other.value
```

Comment:
**suggestion (non-blocking)** could be `return self.name == other.name and self.value == other.value`, but that's not an essential change